Tests/Functional/Drivers/Queue/PhpAmqpLibDriverTest.php

```diff
@@ -0,0 +1,244 @@
+        $this->assertInstanceOf(AsyncQueueConsumer::class, $this->consumer);
+        $this->assertEquals($amqpMessage->delivery_info['consumer_tag'], $this->consumer->getName());
+        $this->assertContains('QueueMessage', $amqpMessage->getBody());
+    }
+
+    /**
+     * Tests the consume process with a callback in return that receives the messages and nack this after
+     *
+     * @dataProvider getMessages
+     * @group amqp-consume-callback-nack
+     */
+    public function testConsumeNackingMessage(MessageInterface $msg)
+    {
+        $this->prepareToConsume($msg);
+        $amqpMessage = null;
+
+        $callback = function($message) use (&$amqpMessage) {
+            $amqpMessage =  $message;
+            $queueMessage = new QueueMessage();
+            $queueMessage->setMessageId($amqpMessage->getDeliveryTag());
+            $this->driver->nack($queueMessage);
+        };
+
+        $this->driver->consume($this->consumer->getName(), $this->queueName, $callback);
+        $this->driver->isConsuming();
+        $this->driver->wait();
+
+        $this->assertInstanceOf(AMQPMessage::class, $amqpMessage);
+        $this->assertEquals($amqpMessage->delivery_info['routing_key'], $this->queueName);
+        $this->assertInstanceOf(AsyncQueueConsumer::class, $this->consumer);
+        $this->assertEquals($amqpMessage->delivery_info['consumer_tag'], $this->consumer->getName());
+        $this->assertContains('QueueMessage', $amqpMessage->getBody());
+    }
+
+    /**
+     * Tests the function to destroy the connection and clean the variables related
+     *
+     * @group destroy
+     */
+    public function testDestroy()
+    {
+        $this->driver->connect();
+        $this->driver->declareChannel();
+        $this->assertTrue($this->driver->isConnected());
+        $this->driver->destroy($this->createConsumer()->getName());
+        $this->assertFalse($this->driver->isConnected());
+    }
+
+     /**
+     * Tests that tries to declare a channel without having a connection
+     *
+     * @group channel-exception
+     */
+    public function testDeclareChannelWithoutConnnection()
+    {
+        $this->expectException(AMQPConnectionClosedException::class);
+        $this->expectExceptionMessage('Broken pipe or closed connection');
+        $this->assertTrue($this->driver->isConnected());
+        $this->driver->declareChannel();
+        $this->driver->destroy($this->createConsumer()->getName());
+        $this->assertFalse($this->driver->isConnected());
+        $this->driver->declareChannel();
+    }
+
+    /**
+     * @group format
+     */
+    public function testFormat()
+    {
+        $format = 'text/plain';
+        $this->driver->setFormat($format);
+        $this->assertEquals($format, $this->driver->getFormat());
+    }
+
+    /**
+     * @group amqp-connection
+     * @throws \AMQPConnectionException
+     */
+    public function testConnectWithoutData()
```

Comment:
Couldn't this scenario be like this?
```
$this->driver->connect(true);
$this->assertTrue($this->driver->isConnected());
$this->assertInstanceOf(AMQPStreamConnection::class, $this->driver->declareChannel()->getConnection());
```
Or which is the purpose of this test?

Reply:
The purpose is to try to connect without valid data. So, I'm passing an empty configuration. The class has a shuffle method, what it means that the driver can try to connect with empty values. Once it fails, the driver must to try to connect with the valid data and return a valid connection.

Reply:
I would like to clarify this personally

Reply:
As discussed, check this
```
$driver = $this->createDriver();
$driver->configure('', '', '', '');
$driver->connect();
$this->assertTrue($driver->isConnected());
$this->assertInstanceOf(AMQPStreamConnection::class, $driver->declareChannel()->getConnection());
```

Reply:
Refactored.

Reply:
:+1: